### PR TITLE
Re-work unit tests for user and users

### PR DESCRIPTION
**Pull request by clintoncwolfe**
This PR overhauls the testing system to rely on a BackendFactory, which is configured without touching the resource constructor.  It also simplifies the unit test layout, using a constructor, recall, properties approach and having explicit test fixture data, rather than setting up mocks.

This is the approach we'll likely be using moving forward; it makes it easy to add new properties and filter behavior.

It also removes the user provider support libraries.

### CODE REVIEW

File: libraries/aws_iam_user.rb [hunk 1]
```diff
@@ -2,57 +2,106 @@
...
+      raise ArgumentError, 'You must provide a username to aws_iam_user.'
+    end
+    validated_params
   end
 
-  def to_s
-    "IAM User #{name}"
+  def fetch_from_aws
+    backend = BackendFactory.create
+    @aws_user_struct ||= nil # silence unitialized warning
```
Comment (jerryaldrichiii) [inline libraries/aws_iam_user.rb]: Thoughts on `unless defined?(aws_user_struct) && aws_user_struct` instead? That wouldn't need a comment explaining it.

File: libraries/aws_iam_users.rb [hunk 2]
```diff
@@ -19,46 +19,84 @@ class AwsIamUsers < Inspec.resource(1)
...
   end
 
   def to_s
     'IAM Users'
   end
 
-  class AwsIamUserFactory
-    def create_user(user)
-      AwsIamUser.new(user: user)
+  # Entry cooker.  Needs discussion.
```
Comment (jerryaldrichiii) [inline libraries/aws_iam_users.rb]: Does this need discussion before merge?

**Review by jerryaldrichiii — APPROVED** [review]
Great contribution! No blocking feedback, just a few questions.

File: libraries/aws_iam_users.rb (hunk 2 above)
Comment (clintoncwolfe) [reply libraries/aws_iam_users.rb]: It's an inspec-project-wide issue.  If you access the entries of a FilterTable-basd resource, should you get back an Inspec::Resource object, or something like a plain hash?

On inspec core, it's generally a plain hash (or something FilterTable-derived that acts like it).  On inspec-aws, we've seen a variety of approaches.  

I think this is an issue for the style guide on inspec core.  For now, this PR makes the aws_iam_users resource act like a typical inspec resource.

That is potentially a breaking change; but this is an incubator project.  I don't want to commit to returning "cooked" (as opposed to "raw") prior to merging into inspec-core.

File: libraries/aws_iam_user.rb (hunk 1 above)
Comment (clintoncwolfe) [reply libraries/aws_iam_user.rb]: To me, the comment seems clearer.  @miah?

**Comment by clintoncwolfe** [comment]
@arlimus, I think this is ready for merge; it's blocking some work from the community on #123

**Review by arlimus — APPROVED** [review]
Thank you for the huge list of improvements here @clintoncwolfe !! Amazing work :)

**Pull request merged**